### ISSUE TITLE
Speaker verification with low confidence

### ISSUE THREAD

**Issue by DieguJota** [NVIDIA/NeMo#5930]
Hello

I am using NeMo for my Portuguese language transcriber and I have great results, far superior to other engines, congratulations on the job.
Now I'm working with NeMo to use voice authentication, in my scenario I'm using phone call audio as a base, in 8K and with background noise. I tested all the pre-trained models that you suggest, the "best" being titanet_large and the result from the same people is giving between 50 to 65% confidence. To test I used the code:

`
speaker_model = nemo_asr.models.EncDecSpeakerLabelModel.from_pretrained(model_name='titanet_large')
speaker_model.verify_speakers('/content/PERSON1-AUDIO1.wav', '/content/PERSON1-AUDIO2.wav')
`

And the sample audios (I can't provide more due to privacy concerns):
[NVIDIA.zip](https://github.com/NVIDIA/NeMo/files/10663690/NVIDIA.zip)

As you suggest, I did a finetunning using the same parameters suggested in the training https://github.com/NVIDIA/NeMo/blob/main/tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb.
I only changed the number of Epochs, placing 250, this training ended after 1 day, the loss started at 13.7 and after 250 Epochs it was at 0.0000508

With this model in hand, I did the same tests, but the result got worse, it only gave high confidence when the person said the same thing said in the other audio. It gave about 90% if both audios had "Hello", for example, otherwise it gave 50%.
For this finetunning, I used approximately 60 hours of phone call audio as a basis, with approximately 1100 different speakers.

I opened an old issue with you about this same problem, but I still didn't see any improvements. I tested what was suggested from 6 seconds upwards and there were no improvements. I'm following titanet's own paper that comments that we should use at least 3 seconds
https://github.com/NVIDIA/NeMo/issues/2719

Now comes my questions:
Is NeMo suitable for my use case? Phone call audios that are in 8K (For training and testing I convert to 16K).
Do I need to test new parameters? What do you suggest for values ​​and what can I adjust.
Does spoken language matter for training? My test/training audios are in Portuguese.

**Comment by nithinraok** [comment]
Thanks for your question. 

`titanet_large` is trained only on the English dataset, so its performance on other languages is not tested yet.  But it can be finetuned for other languages. Looks like with finetuning you are overfitting. 

> Is NeMo suitable for my use case? Phone call audios that are in 8K (For training and testing I convert to 16K). 

 Yes, during training/finetuning upsample to 16kHz
> Do I need to test new parameters? What do you suggest for values ​​and what can I adjust?

These are the general things that you can try for careful finetuning:
* if you are on 2 GPUs, then start with lr of 0.003 (reduce otherwise) with sgd optimizer and cosine annealing scheduler and epochs as 20. Decrease lr logarithmically for future experiments and compare the performance. 
* Considering 3 sec of audio is good enough for training. During testing the longer the audio the better the quality of speaker embeddings
* Since we don't want to overfit, increasing the margin and scale for angular loss are also viable options. scale of 30 or 40 and margin as 0.3 or 0.2 

> Does spoken language matter for training? My test/training audios are in Portuguese.
* It does, based on our embedding analysis that was captured from `titanet_large` they contain both language and speaker information. The same speaker speaking multiple language embeddings were formed as groups of sub-clusters within a large speaker cluster.

**Comment by DieguJota** [comment]
Thank you for your response @nithinraok 

I have started all the recommended tests. I have some more questions:

Considering that the audio recordings are highly noisy due to the call center environment, is it advisable to perform pre-processing of the training audio recordings, such as noise removal, silence removal, and sound equalization? Does the same apply to the test audio recordings?

What is the recommended amount of seconds per speaker in training? Is using a 15-second audio recording per speaker sufficient, or is more time required? Is it more effective to train using 3-second segments or using an entire 15-second audio recording?

For testing purposes, I understand that a 15-second audio recording would be sufficient to compare against another audio recording, correct?

Can NeMo identify if a person is imitating another person's voice or forcing a different voice? Is it necessary to obtain more audio recordings per speaker in this scenario, or is the recommended number sufficient?

**Comment by nithinraok** [comment]
> Considering that the audio recordings are highly noisy due to the call center environment, is it advisable to perform pre-processing of the training audio recordings, such as noise removal, silence removal, and sound equalization? Does the same apply to the test audio recordings? 

yes please make sure you do silence removal during training. While testing if the audio is longer than 5 sec and has a considerable amount of the speaker's voice then silence removal is not required. Noise removal on overlapping voices is not required if a similar noise is seen during training and testing. 

> What is the recommended amount of seconds per speaker in training? Is using a 15-second audio recording per speaker sufficient, or is more time required? Is it more effective to train using 3-second segments or using an entire 15-second audio recording? 

You can train with any length duration of audio, its not model limitation. But we suggest to train with constant length of 3 sec or 5 sec as it makes training efficient. 

> For testing purposes, I understand that a 15-second audio recording would be sufficient to compare against another audio recording, correct? 

Anything above 5 sec is good enough. 

> Can NeMo identify if a person is imitating another person's voice or forcing a different voice? Is it necessary to obtain more audio recordings per speaker in this scenario, or is the recommended number sufficient? 

We don;t provide any out-of-the-box implementation for spooyfing identification. But you can train the model accordingly based on the requirement. Always more data the better.

**Comment by DieguJota** [comment]
Thanks for the reply and for the help. @nithinraok 

Just one more question. Is there a recommendation on how many samples of each speaker I should collect? For example, if I'm recording 5-second audio clips, would 10 samples be enough for training?
